server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -437,27 +440,69 @@ private String sanitizeMatchesFilter(String matches) {
             List<String> entitlementIds = criteria.list();
 
             if (entitlementIds != null && !entitlementIds.isEmpty()) {
-                criteria = this.currentSession()
-                    .createCriteria(Entitlement.class)
-                    .add(CPRestrictions.in("id", entitlementIds));
-
-                entitlementsPage = listByCriteria(criteria, pageRequest);
+                entitlementsPage = createPage(findEntitlements(entitlementIds), pageRequest);
             }
             else {
                 entitlementsPage = new Page<>();
-                entitlementsPage.setPageData(Collections.<Entitlement>emptyList());
+                entitlementsPage.setPageData(Collections.emptyList());
                 entitlementsPage.setMaxRecords(0);
             }
         }
-
         return entitlementsPage;
     }
 
+    private Set<Entitlement> findEntitlements(List<String> entitlementIds) {
```

Comment:
If we're going to be using this lookup for paging, we need the query itself to be ordered, and we cannot use a set (as its ordering is not guaranteed either).

The entire criteria usage itself likely needs to be axed here.

Reply:
Done.